Sources/web3swift/Contract/EthereumContract.swift

```diff
@@ -6,137 +6,140 @@
 import Foundation
 import BigInt
 
-public struct EthereumContract: ContractProtocol {
+public class EthereumContract: ContractProtocol {
+
     public var transactionOptions: TransactionOptions? = TransactionOptions.defaultOptions
     public var address: EthereumAddress? = nil
 
-    var _abi: [ABI.Element]
+    public let abi: [ABI.Element]
 
-    public var allEvents: [String] {
-        return events.keys.compactMap({ (s) -> String in
-            return s
-        })
-    }
+    private(set) public lazy var methods: [String: [ABI.Element.Function]] = {
+        var methods = [String: [ABI.Element.Function]]()
```

Comment:
This looks like computed variable, so there's could not be set modifier, coz we actually can't set it anyhow.

Reply:
Probably not relevant anymore. Please, see the latest version of `EthereumContract`.